### PR TITLE
Remove usages of ServiceContext from Caption component

### LINKED ISSUE #995: Remove ServiceContext in the Caption component
**Is your feature request related to a problem? Please describe.**
ServiceContext.Consumer usage should be in `containers/X`, not `components/X`.
We will be separately adding the ServiceContext into a `contexts/` directory. 

Ideally we would initialise Caption with a `imageCaptionOffscreenText` _prop_, rather than rely on context.

Similar issue to #994.

**Describe the solution you'd like**
- Create a CaptionContainer (inside containers/Figure/Caption)
- Import ServiceContext inside CaptionContainer
- Pass a prop (`imageCaptionOffscreenText`) to the Caption component
- Remove the ServiceContext from the presentational component
- Use CaptionContainer inside FigureContainer

**Describe alternatives you've considered**
N/A

**Testing notes**
Regression testing for the caption within the article. This would be looking at the responsive style, current/future/older versions of web browsers and non-JS behaviour.

Dev insight: Just regression tests needed. Everything should look identical.

**Additional context**
This is a prerequisite for moving to the Psammead component library - we want the Brand component to be as reusable as possible.

- [x] Initially labelled with ["Refinement needed"](https://github.com/BBC-News/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by bcmn**
Resolves #995. 

A small change which ended up pulling in quite a bit more... 😅 This PR:

**Component Layer**
* Simplifies the Figure Caption component entirely, to simply be a styled component
* Updates the Figure Caption component tests to remove all references to ServiceContext, & simply test this new behaviour.
* Removes Text component snapshot test around using Caption as a paragraphOverride [as per this discussion](https://github.com/BBC-News/simorgh/pull/1019#discussion_r237153406)

**Container Layer**
* Logic for determining the offscreen text for Caption was moved to the container layer. In doing so, this increased the amount of container logic specifically for the Caption, so it was pulled out into a new Caption container.
* Visually Hidden Text is handled entirely within the container layer, as there is potential for more complex logic to determine it's position as we support additional languages.
* Adds snapshot tests for the new Caption container, testing under the usage of ServiceContextProvider.

- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: src/app/containers/Caption/index.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { node, string } from 'prop-types';
+import Caption from '../../components/Figure/Caption';
+import Text from '../../components/Text';
+import { ServiceContext } from '../../components/ServiceContext';
+
+const CaptionWithOffscreenText = ({ children }) => (
+  <ServiceContext.Consumer>
+    {({ imageCaptionOffscreenText }) => (
+      <Caption offscreenTextPrefix={imageCaptionOffscreenText}>
```
Comment (pjlee11) [inline src/app/containers/Caption/index.jsx]:
```suggestion
      <Caption>
        <VisuallyHiddenText>{imageCaptionOffscreenText}</VisuallyHiddenText>
```

By doing the above it means the `Caption` component doesn't need to be aware of where the offscreen text is rendered and therefore it is a pure component which renders a `<figcaption>`. This also makes it easier in the future if the offscreen text needs to be a suffix (or both prefix and suffix) depending on the context of the language.

File: src/app/components/Text/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -65,7 +65,7 @@ exports[`Text should render a <figcaption> as per the Caption component override
   <span
     className="c1"
   >
-    Image caption, 
+
```
Comment (pjlee11) [inline src/app/components/Text/__snapshots__/index.test.jsx.snap]: This feels like it shouldn't of changed if the test correctly was passed the context

**Review by pjlee11 — CHANGES_REQUESTED** [review]
One note on how the offscreen text should be injected into the component

File: src/app/components/Text/__snapshots__/index.test.jsx.snap (hunk 2 above)
Comment (bcmn) [reply src/app/components/Text/__snapshots__/index.test.jsx.snap]: The test is for the Text _component_ -- the context logic should no longer really exist at this layer. If wanted, I could change it to basically be an exact copy (sans Contexts) of how the (including duplicating the higher order component CaptionWithOffscreenText), but I wasn't sure that was necessary. Thoughts?

Comment (pjlee11) [reply src/app/components/Text/__snapshots__/index.test.jsx.snap]: If the higher order component was exported could it not be reused by the test (and story if a caption story is needed in text - probably shouldn't have one)? That said, I do get your point that the Text component shouldn't really be aware of the caption. If the Text tests shouldn't know about the Caption then the test should be removed entirely.

Comment (bcmn) [reply src/app/components/Text/__snapshots__/index.test.jsx.snap]: Discussed this in person. We came to the realisation that the Caption container is now testing the paragraph override of Caption anyway, & there is no reason for the Text component to be testing every single variation of it's usage. As such, the snapshot will be removed.

File: src/app/components/Figure/Caption/index.test.jsx [hunk 3]
```diff
@@ -1,27 +1,10 @@
...
-shouldMatchSnapshot(
-  'should render Caption with news service context',
-  CaptionWithContext('This is some Caption text', newsServiceContextStub),
-);
-
 shouldMatchSnapshot(
-  'should render Caption with persian service context',
-  CaptionWithContext('توصیف چیزی که اتفاق می افتد', persianServiceContextStub),
+  'should render Caption with some offscreen text',
+  <Caption offscreenTextPrefix="Some offscreenn text">
```
Comment (ChrisBAshton) [inline src/app/components/Figure/Caption/index.test.jsx]: `offscreenn` -> `offscreen`

**Review by ChrisBAshton — CHANGES_REQUESTED** [review]
Good work Ben - I echo what Phil said, let's make `Caption` naïvely render `props.children` rather than having to know about rendering the `VisuallyHiddenText` component specifically. (Also makes it easier to add changes when we get the new caption designs from UX).

Not sure how difficult that is in the one PR - would need to make changes to the Text component, I think...? 🤔Worth seeing how far you can get with it.

**Comment by bcmn** [comment]
Thanks for the feedback, guys -- updated.

**Review by pjlee11 — DISMISSED** [review]

**Review by sareh — COMMENTED** [review]
👍 Looks good!

File: src/app/containers/Caption/index.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import Caption from '../../components/Figure/Caption';
+import Text from '../../components/Text';
+import { ServiceContext } from '../../components/ServiceContext';
+import VisuallyHiddenText from '../../components/VisuallyHiddenText';
+
+const CaptionWithOffscreenText = ({ children }) => (
+  <ServiceContext.Consumer>
+    {({ imageCaptionOffscreenText }) => (
+      <Caption>
+        <VisuallyHiddenText>{imageCaptionOffscreenText}</VisuallyHiddenText>
```
Comment (sareh) [inline src/app/containers/Caption/index.jsx]: We should conditionally render `<VisuallyHiddenText>`, so if `imageCaptionOffscreenText` is undefined, we don't render and empty span here.  We should definitely have a unit test to capture this usecase.

**Review by sareh — CHANGES_REQUESTED** [review]

**Review by ChrisBAshton — DISMISSED** [review]
This looks like an improvement to me. Good job! 👍

File: src/app/containers/Caption/index.jsx (hunk 4 above)
Comment (bcmn) [reply src/app/containers/Caption/index.jsx]: Done, & as discussed in person, we've added `imageCaptionOffscreenText` properties to the News & Persian service configs.

**Review by sareh — DISMISSED** [review]
👍 Looks great!

**Review by pjlee11 — APPROVED** [review]
Nice work 😄

**Review by sareh — APPROVED** [review]

**Comment by bcmn** [comment]
Now blocking #1027.

**Comment by jamesbrumpton** [comment]
LGTM

**Pull request merged**